client/src/components/container/portal/Attendance/index.js

```diff
@@ -66,29 +74,32 @@ class Attendance extends React.Component {
     error(err);
   };
 
-  handleModalSubmit = ({
+  handleModalSubmit = async ({
     target: {
       parentNode: {
         firstChild: { value },
       },
     },
   }) => {
     const { success, error } = this;
-    axios
-      .post('/api/v1/portal/logout', { pinCode: value })
-      .then(({ data: { msg } }) => {
-        success(msg);
-        this.props.history.push('/portal');
-      })
-      .catch(
-        ({
-          response: {
-            data: { msg },
-          },
-        }) => {
-          error(msg);
-        }
-      );
+    const {
+      props: {
+        history: { push },
+      },
+    } = this;
+    try {
+      const {
+        data: { msg },
+      } = await axios.post('/api/v1/portal/logout', { pinCode: value });
+      success(msg);
+      push('/portal');
+    } catch ({
+      response: {
+        data: { msg },
+      },
+    }) {
+      error(msg);
+    }
```

Comment:
edit error handling here as the previous comment : 
```suggestion
       catch (err) {
      let errorMesg;
      if (err.response) {
        errorMesg = err.response.data.msg;
      } else {
        errorMesg = 'Something went wrong, please try again later';
      }
      error(errorMesg);
    }
```